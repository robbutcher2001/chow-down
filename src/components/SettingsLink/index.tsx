import React, { ReactNode } from 'react';
import { Link } from 'react-router-dom';

import styled from 'styled-components';

interface SettingsLinkProps {
  to: string,
  children: ReactNode
};

const Navigation = styled.nav`
  display: flex;
  justify-content: center;
  margin-top: 1.5rem;
  font-size: ${props =>
    props.theme.typography.fontSize.large
  };

  a {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    max-width: 500px;
    color: black;
    text-decoration: none;
    border: solid #4acaa8;
    border-width: 0 0 3px 0;
    padding: 1rem;
  }

  i {
    border: solid black;
    border-width: 0 2px 2px 0;
    display: inline-block;
    padding: 6px;
    transform: rotate(-45deg);
  }
`

export default (props: SettingsLinkProps) => (
  <Navigation>
    <Link to={props.to}>
      {props.children}
      <i/>
    </Link>
  </Navigation>
);