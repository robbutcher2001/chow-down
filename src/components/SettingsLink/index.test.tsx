import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import SettingsLink from '.';

test('SettingsLink basic snapshot render', () => {
  const settingsLink = renderer.create(
    <Router history={createMemoryHistory()}>
      <SettingsLink to='fake_link' >
        <div>fake_child</div>
      </SettingsLink>
    </Router>
  );

  expect(settingsLink.toJSON()).toMatchSnapshot();
});

test('SettingsLink correct tag content assertion', () => {
  const { getByText } = render(
    <Router history={createMemoryHistory()}>
      <SettingsLink to='fake_link' >
        <div>fake_child</div>
      </SettingsLink>
    </Router>
  );

  expect(getByText(/fake_child/).textContent).toEqual('fake_child');
});