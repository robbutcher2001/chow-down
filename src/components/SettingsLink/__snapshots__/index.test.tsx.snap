// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsLink basic snapshot render 1`] = `
<nav
  className="sc-AxjAm ddLWSW"
>
  <a
    href="/fake_link"
    onClick={[Function]}
  >
    <div>
      fake_child
    </div>
    <i />
  </a>
</nav>
`;
